application/forms/EventRuleConfigForm.php

```diff
@@ -628,6 +642,33 @@ public function removeRule(int $id): void
         $db->commitTransaction();
     }
 
+    /**
+     * Get the newly made changes
+     *
+     * @return array
+     */
+    public function getChanges(): array
+    {
+        $values = $this->getValues();
+        $dbValuesToCompare = array_intersect_key($this->config, $values);
+
+        $checker = static function ($a, $b) use (&$checker) {
+            if (! is_array($a) || ! is_array($b)) {
+                return $a <=> $b;
+            }
+
+            return empty(array_udiff_assoc($a, $b, $checker)) ? 0 : 1;
+        };
+
+        $diff = array_udiff_assoc($values, $dbValuesToCompare, $checker);
+
+        // Search editor is not part of the form and the filter is dynamically updated in the config after its
+        // submission. Hence, it must be part of the changes.
+        $diff['object_filter'] = $this->config['object_filter'] ?? null;
```

Comment:
Now the returned value is never empty, which leads to false positives.
Do not edit anything and click on `Save`, then the success message will appear.

Reply:
I am aware of this. But since, there was no priority on the PR. I did not work on this issue. I will fix it.